### ISSUE TITLE
private builder mode 

### ISSUE THREAD

**Issue by hiaselhans** [gumyr/build123d#141]
I tried to split a builder into pieces by building new compound classes and using them at the desired locations.
The docs advice to use a private builder context and in that context i would expect to start with a blank location (set at origin). Instead Locations from an outer scope shadow into the new context.

Here is a minimal example:

``` python
def nut():
    with bd.BuildPart(mode=bd.Mode.PRIVATE) as part:
        with bd.BuildSketch():
            bd.RegularPolygon(4, 6)
        
        bd.Extrude(amount=5)
    
    return part.part



with bd.BuildPart() as part:
    with bd.Locations((-10,-10), (10,10), (-10,10), (10,-10)) as locations:
        bd.Add(nut())

part
```
expected result:
![image](https://user-images.githubusercontent.com/5046381/219961179-40d31344-e6a2-43a3-9151-17fb5097f101.png)

actual result:
![image](https://user-images.githubusercontent.com/5046381/219961148-b90b95cb-ef2b-447b-bdd2-ba632c8047ea.png)

**Comment by gumyr** [comment]
The global Locations in BuildPart are interacting with the code in the function.  This:
```py
def nut():
    with bd.BuildPart(bd.Plane.XY, mode=bd.Mode.PRIVATE) as part:
        with bd.BuildSketch():
            bd.RegularPolygon(4, 6)

        bd.Extrude(amount=5)

    return part.part
```
produces this:
![image](https://user-images.githubusercontent.com/85638518/219974136-f68932f1-b39a-403e-85ca-8d6218f785dd.png)
Note that I'm setting the workplane and therefore the default Location.

I've worked on trying to automate this for quite a while and been unable to come up with a solution.  The requirement to set the workplane and mode in functions and classes will be documented here: https://build123d.readthedocs.io/en/latest/custom.html when I get a chance.

**Comment by gumyr** [comment]
Fixed with commit 0ae15a463be547ab73fa24559d61602e7c4a6e3e
